File: ToolTutorials/UnitTestFramework/Box3D.py
# 3D Box class
# Author: Mustafa Haddara

# This class represents a rectangle as (x,y,z) coordinate representing the 
# front-facing top-left corner, and a (w,h,d) tuple representing the width,
# height, depth
class Box3D(object):

    # Faces enums
    FRONT = 0
    TOP = 1
    SIDE = 2

    # Constructor accepts 2 parameters
    # corner: (x,y,z) tuple for the top left corner
    # size: (w,h,d) tuple containing the width and height
    def __init__(self, corner, size):
        if len(corner) != 3 or len(size) != 3:
            raise ValueError("Invalid argument size")
        # double underscore (__) before the identifier is python convention
        # for private variables
        self.__frontTopLeftCorner = corner
        self.__dimensions = size

    def getFrontTopLeftCorner(self):
        return self.__frontTopLeftCorner

    def setFrontTopLeftCorner(self, newCorner):
        self.__frontTopLeftCorner = newCorner

    def getDimensions(self):
        return self.__dimensions

    def setDimensions(self, newSize):
        self.__dimensions = newSize

    def getDimensionsOfFace(self, face):
        if face == Box3D.FRONT:
            width,height = self.__dimensions[0],self.__dimensions[1] # w,h
        elif face == Box3D.TOP:
            width,height = self.__dimensions[0],self.__dimensions[2] # w,d
        elif face == Box3D.SIDE:
            width,height = self.__dimensions[1],self.__dimensions[2] # h,d
        else:
            raise ValueError("Invalid `face` argument")
        return width,height

    def getAreaOfFace(self, face):
        width,height = self.getDimensionsOfFace(face)
        return width * height

    def getPerimeterOfFace(self, face):
        width,height = self.getDimensionsOfFace(face)
        return 2 * (width + height)

    def getVolume(self):
        width,height,depth = self.__dimensions
        return width * height * depth

    def getSurfaceArea(self):
        surfaceArea = 0
        for face in (Box3D.FRONT, Box3D.TOP, Box3D.SIDE):
            surfaceArea += (2 * self.getAreaOfFace(face))
        return surfaceArea

# Example usage
if __name__ == '__main__':
    b = Box3D( (1,1,1), (2,2,2) )
    print b.getSurfaceArea()

File: ToolTutorials/UnitTestFramework/Box3D.java
/* 3D Box class
 * Author: Mustafa Haddara
 *
 * This class represents a rectangle as (x,y,z) coordinate representing the 
 * front-facing top-left corner, and a (w,h,d) tuple representing the width,
 * height, depth
 */
public class Box3D {

    // Faces enums
    enum Face {
        FRONT,
        TOP,
        SIDE
    }

    private int[] frontTopLeftCorner;
    private int[] dimensions;

    /* Constructor accepts 2 parameters
     * corner: (x,y,z) tuple for the top left corner
     * size: (w,h,d) tuple containing the width and height
     */
    public Box3D (int[] corner, int[] size) {
        if(corner.length != 3 || size.length != 3) {
            throw new RuntimeException("Invalid argument size");
        }
        this.frontTopLeftCorner = corner;
        this.dimensions = size;
    }

    public int[] getFrontTopLeftCorner() {
        return this.frontTopLeftCorner;
    }

    public void setFrontTopLeftCorner(int[] newCorner) {
        this.frontTopLeftCorner = newCorner;
    }

    public int[] getDimensions() {
        return this.dimensions;
    }

    public void setDimensions(int[] newSize) {
        this.dimensions = newSize;
    }

    public int[] getDimensionsOfFace(Face face) {
        int[] dim = new int[3];
        if (face == Face.FRONT) {
            dim = new int[] { this.dimensions[0], this.dimensions[1] };
        } else if (face == Face.TOP) {
            dim = new int[] { this.dimensions[0], this.dimensions[2] };
        } else if (face == Face.SIDE) {
            dim = new int[] { this.dimensions[1], this.dimensions[2] };
        }

        return dim;
    }

    public int getAreaOfFace(Face face) {
        int[] dim = this.getDimensionsOfFace(face);
        return dim[0] * dim[1];
    }

    public int getPerimeterOfFace(Face face) {
        int[] dim = this.getDimensionsOfFace(face);
        return 2 * (dim[0] + dim[1]);
    }

    public int getVolume() {
        int width  = this.dimensions[0];
        int height = this.dimensions[1];
        int depth  = this.dimensions[2];
        return width * height * depth;
    }

    public int getSurfaceArea() {
        int surfaceArea = 0;
        for (Face face: Face.values()) {
           surfaceArea += (2* this.getAreaOfFace(face));
        }
        return surfaceArea;
    }

    // Example usage
    public static void main(String[] args) {
        Box3D b = new Box3D( new int[] {1,1,1}, new int[] {2,2,2});
        System.out.println(b.getVolume());
        System.out.println(b.getSurfaceArea());
    }
}